### PR TITLE
completed #5

### PR DESCRIPTION
**Pull request by pkaodev**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -62,13 +62,99 @@ describe('GET/api/reviews/:review_id', () => {
...
+            created_at: "2021-01-18T10:00:20.514Z",
+            votes: 1
+        }};
+
+        return request(app).patch('/api/reviews/1').send(sentObj).expect(200)
+        .then(response => {
+            expect(response.body).toEqual(expectedObj);
+        })
+    });
+    it('404: "404: ID Out Of Range" when given out of range number for :review_id', () => {
```
[the remark this answers is not in the document]
Comment (Rosem10) [reply __tests__/app.test.js:139]: I'd think again about how you've phrased this as "out of range" - what if, later, you create an endpoint to delete reviews, and I come skipping along and delete, say, id number 9? It should still give a 404, so the "range" element is a bit misleading.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -62,13 +62,99 @@ describe('GET/api/reviews/:review_id', () => {
...
             expect(JSON.parse(response.text)).toEqual({msg: '400: Invalid ID Format'})
         })
     });
+    it('400: "400: Invalid Vote Format" when given any non-integer in request object ', () => {
+        const sentObj = {inc_votes: 'like a string'};
+        return request(app).patch('/api/reviews/1').send(sentObj).expect(400)
+        .then(response => {
+            expect(JSON.parse(response.text)).toEqual({msg: '400: Invalid Vote Format'})
+        })
+    });
```
Comment (Rosem10) [inline __tests__/app.test.js:159]: One more test for if the sentObj is empty, please!

File: models/reviews.model.js [hunk 3]
```diff
@@ -1,22 +1,44 @@
 const db = require('../db/connection')
 
 //#4
-exports.fetchReviewById = (reviewId) => {
+exports.fetchReviewById = ({review_id}) => {
 
     //reject promise if :review_id contains non-numbers
-    if (/[^\d]/.test(reviewId.review_id)) {
+    if (/[^\d]/.test(review_id)) {
```
Comment (Rosem10) [inline models/reviews.model.js:7]: Impressed that you got regex in here - but the promise below will automatically fail if given the wrong type (pg will be expecting an integer for the id, so anything else will throw an error)

File: models/reviews.model.js [hunk 4]
```diff
@@ -1,22 +1,44 @@
...
         }
 
+        return result.rows[0];
+    })
+}
+
+//#5
+updateReviewVotes = ({review_id}, {inc_votes}) => {
+    //reject promise if :review_id contains non-numbers
+    if (/[^\d]/.test(review_id)) {
```
Comment (Rosem10) [inline models/reviews.model.js:27]: Same again, these will automatically fail anyhow once pg gets to them below, so you can save yourself a few lines of code.

File: models/reviews.model.js [hunk 5]
```diff
@@ -1,22 +1,44 @@
...
+    }
+
+
+    return db.query('SELECT * FROM reviews WHERE review_id = $1', [review_id])
+    .then(result => {
+        //reject promise if :review_id is out of range (result array is empty)
+        if (result.rows.length === 0) {
+            return Promise.reject({status: 404, msg: '404: ID Out Of Range'});
+        }
+        result.rows[0].votes += inc_votes;
```
Comment (Rosem10) [inline models/reviews.model.js:41]: I see what you've done here, but it's not quite right. Whilst what you are returning to the user/tests is showing an increased vote, this won't actually be reflected in our database. Take another look at your pg query to think about how we can get it to update the vote count of the relevant review in the db, please!

**Review by Rosem10 — CHANGES_REQUESTED** [review]
Hometime wooo!

File: __tests__/app.test.js [hunk 6]
```diff
@@ -62,13 +62,99 @@ describe('GET/api/reviews/:review_id', () => {
...
         return request(app).get('/api/reviews/12345').expect(404)
         .then(response => {
-            expect(JSON.parse(response.text)).toEqual({msg: '404: ID Out Of Range'})
+            expect(JSON.parse(response.text)).toEqual({msg: '404: ID Out Of Range'});
         })
     });
     it('400: "400: Invalid ID Format" when given any non-number character for :review_id', () => {
         return request(app).get('/api/reviews/notanumber').expect(400)
+        .then(response => {
+            expect(JSON.parse(response.text)).toEqual({msg: '400: Invalid ID Format'});
```
[the remark this answers is not in the document]
Comment (pkaodev) [reply __tests__/app.test.js:71]: Thank for all the very helpful comments!!!
-removed doubling up of status+message

File: __tests__/app.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pkaodev) [reply __tests__/app.test.js:139]: Rephrased a few messages to generic 'Invalid Input'

File: __tests__/app.test.js (hunk 2 above)
Comment (pkaodev) [reply __tests__/app.test.js:159]: Added

File: models/reviews.model.js (hunk 3 above)
Comment (pkaodev) [reply models/reviews.model.js:7]: Removed these and added a PG error handler in app.js

File: models/reviews.model.js (hunk 5 above)
Comment (pkaodev) [reply models/reviews.model.js:41]: Bit of a brainfart there!  Have updated sql statement
